core/src/main/java/org/mule/runtime/core/api/util/ClassUtils.java

```diff
@@ -372,36 +379,96 @@ public static Class loadClass(final String className, final ClassLoader classLoa
+    Field field = getField(targetClass, fieldName, recursive);
+    if (!Modifier.isStatic(field.getModifiers())) {
+      throw new IllegalAccessException(String.format("The %s field of %s class is not static", fieldName, targetClass.getName()));
     }
+    field.setAccessible(true);
+    return (T) field.get(null);
+  }
 
-    throw new NoSuchFieldException(String.format("Could not find field '%s' in class %s", fieldName,
-                                                 target.getClass().getName()));
+  /**
+   * Sets a field of an object with the given value. If this is a final field, there will be
+   * an attempt to update the value.
+   * Notice: If the field is final and it was initialized using a constant the value change
+   * may not be reflected in due compiler optimizations.
+   * http://java.sun.com/docs/books/jls/third_edition/html/memory.html#17.5.3
+   * @param target the object that holds the target field
+   * @param fieldName the name of the field
+   * @param value the value to set
+   * @param recursive flags to lookup the field in subclasses of the target object
+   * @throws IllegalAccessException the field is not reachable
+   * @throws NoSuchFieldException the field does not exists.
+   */
+  public static void setFieldValue(Object target, String fieldName, Object value, boolean recursive)
+      throws IllegalAccessException, NoSuchFieldException {
+    Field field = getField(target.getClass(), fieldName, recursive);
+    field.setAccessible(true);
+    if (Modifier.isFinal(field.getModifiers())) {
+      Field modifiersField = getField(Field.class, "modifiers", false);
+      modifiersField.setAccessible(true);
+      modifiersField.setInt(field, field.getModifiers() & ~Modifier.FINAL);
+    }
+    field.set(target, value);
   }
 
+  /**
+   * Sets a static field of a given class, even if the field has the final modifier.
+   * Notice: If the field is final and it was initialized using a constant the value change
+   *         may not be reflected due compiler optimizations.
+   *         http://java.sun.com/docs/books/jls/third_edition/html/memory.html#17.5.3
+   *
+   * @param targetClass the target class
+   * @param fieldName the name of the field
+   * @param value the value to set
+   * @param recursive flags to lookup the field in subclasses or not
+   * @throws NoSuchFieldException when the field does not exists
+   * @throws IllegalAccessException when the field is not reachable
+   */
+  public static void setStaticFieldValue(Class<?> targetClass, String fieldName, Object value, boolean recursive)
+      throws NoSuchFieldException, IllegalAccessException {
+    Field field = getField(targetClass, fieldName, recursive);
+    if (Modifier.isFinal(field.getModifiers())) {
+      Field modifiersField = getField(Field.class, "modifiers", false);
+      modifiersField.setAccessible(true);
+      modifiersField.setInt(field, field.getModifiers() & ~Modifier.FINAL);
+    }
+    field.setAccessible(true);
```

Comment:
set accessible to its original value after getting the value of the field, in a finally block